aten/src/ATen/test/cuda_stream_test.cpp

```diff
@@ -256,3 +257,23 @@ TEST(TestStream, CrossDeviceTest) {
   cudaStreamSynchronize(stream0);
   ASSERT_TRUE(event0.query());
 }
+
+// Generic Events
+TEST(TestStream, GenericEventTest) {
+  if (!at::cuda::is_available()) return;
+
+  c10::Event event{};
+  c10::Stream stream = at::cuda::getStreamFromPool();
+
+  event.recordOnce(stream);
+
+  const c10::Stream wait_stream0 = at::cuda::getStreamFromPool();
+  const c10::Stream wait_stream1 = at::cuda::getStreamFromPool();
+
+  event.block(wait_stream0);
```

Comment:
I suppose our lower level APIs are already this way, but I find `stream.wait(event)` much clearer than `event.block(stream)`. I am not sure what are reasons are for having the `block` but not the the `wait`.

Reply:
I think we can add stream.wait(event) in this PR, too. Pseudocode: 

`Stream::wait(event) = event.block(this)`

Reply:
Added this (tested in next test).